_posts/2017-03-06-gcp-dataflow-streaming-example-project-0.1.0-released.md

```diff
@@ -0,0 +1,238 @@
+
+<div class="html">
+<h2><a name="introducing-analytics-on-write">2. Introducing analytics-on-write</a></h2>
+</div>
+
+Snowplow being an event data pipeline, real-time processing of event streams is at its heart. This project can be seen as a simplification of what happens in a real life event data pipeline - one that, for example, does real-time processing of events relative to temperature drops and rises.
+Our Dataflow job reads a Pub/Sub topic containing events in a JSON format:
+
+{% highlight json %}
+{
+  "timestamp": "2015-06-05T12:54:43.064528",
+  "type": "Green",
+  "id": "4ec80fb1-0963-4e35-8f54-ce760499d974"
+}
+{% endhighlight %}
+
+Our job counts the events by `type` and aggregates these counts into 1 minute buckets. The job then takes these aggregates and stores them into a table in Cloud Bigtable.
+
+1. **Downsampling:** where we reduce the event's ISO 8601 timestamp down to minute precision, so for instance "2015-06-05T12:54:43.064528" becomes "2015-06-05T12:54:00.000000". This downsampling gives us a fast way of bucketing or aggregating events via this downsampled key
+2. **Bucketing:** an aggregation technique that builds buckets, where each bucket is associated with a downsampled timestamp key and an event type criterion. By the end of the aggregation process, we’ll end up with a list of buckets - each one with a countable set of events that "belong" to it.
+
+<div class="html">
+<h2><a name="detailed-setup">3. Detailed setup</a></h2>
+</div>
+
+In this tutorial, we'll walk through the process of getting up and running with GCP Pub/Sub and Dataflow. You will need  [git] [git-install], [Vagrant] [vagrant-install] and [VirtualBox] [virtualbox-install] installed locally. 
+
+<h3>Step 1: Build the project</h3>
+
+In your local terminal:
+
+{% highlight bash %}
+ host$ git clone https://github.com/snowplow/google-cloud-dataflow-example-project
+ host$ cd google-cloud-dataflow-example-project
+ host$ vagrant up && vagrant ssh
+{% endhighlight %}
+
+Let's now build the project. This should take around 10 minutes with these commands:
+
+{% highlight bash %}
+guest$ cd /vagrant
+guest$ inv build_project
+{% endhighlight %}
+
+<h3>Step 2: Setup project and authenticate</h3>
+
+You're going to need credentials to run certain parts of this project. 
+You'll also need to create a new project in GCP and enable some APIs/services.
+
+(Here's a [guide](https://github.com/snowplow/snowplow/wiki/GCP:-Getting-Started) on Snowplow's wiki on how to setup a GCP project and enable APIs/services)
+
+- Cloud Dataflow
+- Cloud Pub/Sub
+- Cloud Storage
+- Cloud Bigtable
+
+After enabling these services, authenticate with your GCP account by running the following commands:
+
+{% highlight bash %}
+guest$ gcloud auth login
+guest$ gcloud auth application-default login
+guest$ gcloud config set project <your-project-id>
```

Comment:
`<your-project-id>` could be `$YOUR_PROJECT_ID`. Capital letters are more notable and users may want to copy-paste it to shell script.